services/meta/client.go

```diff
@@ -1185,6 +1155,21 @@ func (c *Client) retryUntilSnapshot(idx uint64) *Data {
 	}
 }
 
+func (c *Client) updateAuthCache() {
+	// copy cached user info for still-present users
+	newCache := make(map[string]authUser, len(c.authCache))
```

Comment:
Yes, but we're also reallocating the entire  meta.Data object on each
snapshot...

Since the user information is stored in a slice, we would have to traverse
the map, and then traverse the slice looking for deleted users. That's why
I decided to just copy the relevant cache entries to a new map.
On Feb 11, 2016 6:41 PM, "Jason Wilder" <notifications@github.com> wrote:

> In services/meta/client.go
> <https://github.com/influxdata/influxdb/pull/5644#discussion_r52686323>:
>
> > @@ -1185,6 +1155,21 @@ func (c *Client) retryUntilSnapshot(idx uint64) *Data {
> >  	}
> >  }
> >
> > +func (c *Client) updateAuthCache() {
> > +	// copy cached user info for still-present users
> > +	newCache := make(map[string]authUser, len(c.authCache))
>
> Won't this end up allocating a new map everytime the snapshot is updated
> even if there are no changes to the auth cache?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/influxdata/influxdb/pull/5644/files#r52686323>.
>